qutip/core/environment.py

```diff
@@ -0,0 +1,2017 @@
+
+
+def _complex_interpolation(fun, xlist, name):
+    if callable(fun):
+        return fun
+    else:
+        real_interp = _real_interpolation(np.real(fun), xlist, name)
+        imag_interp = _real_interpolation(np.imag(fun), xlist, name)
+        return lambda x: real_interp(x) + 1j * imag_interp(x)
+
+
+def _fft(f, wMax, tMax):
+    r"""
+    Calculates the Fast Fourier transform of the given function. We calculate
+    Fourier transformations via FFT because numerical integration is often
+    noisy in the scenarios we are interested in.
+
+    Given a (mathematical) function `f(t)`, this function approximates its
+    Fourier transform
+
+    .. math::
+        g(\omega) = \int_{-\infty}^\infty dt\, e^{-i\omega t}\, f(t) .
+
+    The function f is sampled on the interval `[-tMax, tMax]`. The sampling
+    discretization is chosen as `dt = pi / (2*wMax)` (Shannon-Nyquist + some
+    leeway). However, `dt` is always chosen small enough to have at least 250
+    samples on the interval `[-tMax, tMax]`.
+
+    Parameters
+    ----------
+    wMax: float
+        Maximum frequency of interest
+    tMax: float
+        Support of the function f (i.e., f(t) is essentially zero for
+        `|t| > tMax`).
+
+    Returns
+    -------
+    The fourier transform of the provided function as an interpolated function.
+    """
+    # Code adapted from https://stackoverflow.com/a/24077914
+
+    numSamples = int(
+        max(250, np.ceil(4 * tMax * wMax / np.pi + 1))
+    )
+    t, dt = np.linspace(-tMax, tMax, numSamples, retstep=True)
+    f_values = f(t)
+
+    # Compute Fourier transform by numpy's FFT function
+    g = np.fft.fft(f_values)
+    # frequency normalization factor is 2 * np.pi / dt
+    w = np.fft.fftfreq(numSamples) * 2 * np.pi / dt
+    # In order to get a discretisation of the continuous Fourier transform
+    # we need to multiply g by a phase factor
+    g *= dt * np.exp(1j * w * tMax)
+
+    return _complex_interpolation(
+        np.fft.fftshift(g), np.fft.fftshift(w), 'FFT'
+    )
+
+
+def _cf_real_fit_model(tlist, a, b, c, d=0):
+    return np.real((a + 1j * d) * np.exp((b + 1j * c) * np.abs(tlist)))
+
+
+def _cf_imag_fit_model(tlist, a, b, c, d=0):
+    return np.sign(tlist) * np.imag(
+        (a + 1j * d) * np.exp((b + 1j * c) * np.abs(tlist))
+    )
+
+
+def _default_guess_cfreal(tlist, clist, full_ansatz):
+    corr_abs = np.abs(clist)
+    corr_max = np.max(corr_abs)
+    t0 = tlist[np.argmax(corr_abs)]
+
+    # Checks if constant array, and assigns zero
+    if (clist == clist[0]).all():
+        if full_ansatz:
+            return [[0] * 4]*3
+        return [[0] * 3]*3
+
+    if full_ansatz:
+        lower = [-100 * corr_max, -np.inf, -1, -100 * corr_max]
+        guess = [corr_max, -100*corr_max, 0, 0]
+        upper = [100*corr_max, 0, 1, 100*corr_max]
+    else:
+        lower = [-20 * corr_max, -np.inf, 0]
+        guess = [corr_max, -t0, t0]
+        upper = [20 * corr_max, 0.1, np.inf]
```

Comment:
did you have some intuition for using t0 as a guess here?   for some examples I tried on monday it would make the the fit unfeasible

Reply:
Didn't notice $w_{c}$ got renamed to $t_{0}$. The intuition, is that the parameter corresponds to the inverse of the cutoff time (inverse of the cutoff frequency). Not sure why it would be  unfeasible, I think it should always be feasible. Unless it was with negative times for which the model functions had divergent exponents, which should be fixed now ! 

But if you share one of those examples I can take a look!